lucille-core/src/main/java/com/kmwllc/lucille/stage/AddRandomField.java

```diff
@@ -45,31 +41,36 @@
  * <b>maxNumOfTerms</b> (Integer, Optional) : maximum number of terms to be in the field, defaults to 1
  * </p>
  * <p>
- * <b>isNested</b> (FieldType, Optional) : setting for structure of field, default or nested
+ * <b>isNested</b> (bool, Optional) : setting for structure of field, default or nested
+ * <p>
+ * <b>concatenate</b> (bool, Optional) : if true, represent multiple terms as a single space-separated string instead of multiple
+ *  values, defaults to false
  * </p>
  */
 public class AddRandomField extends Stage {
 
+  private static final ObjectMapper MAPPER = new ObjectMapper();
+
   private final String inputDataPath;
   private final String fieldName;
   private Integer rangeSize;
   private Integer minNumOfTerms;
   private Integer maxNumOfTerms;
   private final boolean isNested;
-  private List<String> dataArr;
-  private List<String> uniqueValues;
+  private List<String> fileData;
+  private final boolean concatenate;
 
   public AddRandomField(Config config) throws StageException {
     super(config, new StageSpec().withOptionalProperties("input_data_path", "field_name", "range_size", "min_num_of_terms",
```

Comment:
field_name should be a required field

Reply:
existing behavior is to default to "data" if field_name is not provided:
`    this.fieldName = ConfigUtils.getOrDefault(config, "field_name", "data");`